libs/utils/samplerum.js

```diff
@@ -47,8 +47,11 @@ export function sampleRUM(checkpoint, data = {}) {
       window.hlx.rum = { weight, id, random, isSelected, firstReadTime, sampleRUM, sanitizeURL: urlSanitizers[window.hlx.RUM_MASK_URL || 'path'], rumSessionStorage };
     }
 
-    const { weight, id, firstReadTime } = window.hlx.rum;
-    if (window.hlx && window.hlx.rum && window.hlx.rum.isSelected) {
+    const { id, firstReadTime } = window.hlx.rum;
+    const weight404 = 10;
+    const isSelected404 = checkpoint === '404' && (Math.random() * weight404 < 1);
+    const weight = isSelected404 ? weight404 : window.hlx.rum;
+    if (window.hlx && window.hlx.rum && (window.hlx.rum.isSelected || isSelected404)) {
```

Comment:
On L50, you're directly using `window.hlx.rum`, but here you're making a check for it. Shouldn't we fail faster if it doesn't exist?

Reply:
Since the data is always set on L47, we don't need the safeguard at all.